packages/block-editor/src/components/url-input/index.js

```diff
@@ -22,229 +22,211 @@ import { withInstanceId, withSafeTimeout, compose } from '@wordpress/compose';
 import { withSelect } from '@wordpress/data';
 import { isURL } from '@wordpress/url';
 
-class URLInput extends Component {
-	constructor( props ) {
-		super( props );
-
-		this.onChange = this.onChange.bind( this );
-		this.onFocus = this.onFocus.bind( this );
-		this.onKeyDown = this.onKeyDown.bind( this );
-		this.selectLink = this.selectLink.bind( this );
-		this.handleOnClick = this.handleOnClick.bind( this );
-		this.bindSuggestionNode = this.bindSuggestionNode.bind( this );
-		this.autocompleteRef = props.autocompleteRef || createRef();
-		this.inputRef = createRef();
-		this.updateSuggestions = debounce(
-			this.updateSuggestions.bind( this ),
-			200
-		);
-
-		this.suggestionNodes = [];
-
-		this.isUpdatingSuggestions = false;
-
-		this.state = {
-			suggestions: [],
-			showSuggestions: false,
-			selectedSuggestion: null,
-
-			suggestionsListboxId: '',
-			suggestionOptionIdPrefix: '',
-		};
-	}
-
-	componentDidUpdate( prevProps ) {
-		const { showSuggestions, selectedSuggestion } = this.state;
-		const { value } = this.props;
-
-		// only have to worry about scrolling selected suggestion into view
-		// when already expanded
+const REQUEST_DEBOUNCE_DELAY = 200;
+
+function URLInput( {
+	value,
+	onChange: parentOnChange,
+	label = '',
+	className = '',
+	isFullWidth,
+	instanceId,
+	placeholder = __( 'Paste URL or type to search' ),
+	disableSuggestions,
+	speak,
+	debouncedSpeak,
+	__experimentalRenderControl: experimentalRenderControl,
+	__experimentalRenderSuggestions: experimentalRenderSuggestions,
+	__experimentalFetchLinkSuggestions: fetchLinkSuggestions,
+	__experimentalHandleURLSuggestions: handleURLSuggestions = false,
+	__experimentalShowInitialSuggestions: showInitialSuggestions = false,
+	setTimeout,
+	parentAutocompleteRef,
+} ) {
+	const [ loading, setLoading ] = useState( false );
+	const [ suggestions, setSuggestions ] = useState( [] );
+	const [ showSuggestions, setShowSuggestions ] = useState( false );
+	const [ selectedSuggestion, setSelectedSuggestion ] = useState( null );
+	const [ prevInstanceId, setPrevInstanceId ] = useState( null );
+	const [ suggestionsListboxId, setSuggestionsListboxId ] = useState( '' );
+	const [ suggestionOptionIdPrefix, setSuggestionOptionIdPrefix ] = useState(
+		''
+	);
+	const [ prevDisableSuggestions, setPrevDisableSuggestions ] = useState(
+		null
+	);
+
+	const autocompleteRef = useRef( parentAutocompleteRef );
+	const inputRef = useRef( null );
+	const prevValue = useRef( value );
+	const pendingRequest = useRef( null );
+	const scrollingIntoView = useRef( false );
+	const suggestionNodes = useRef( [] );
+
+	useEffect( () => {
+		prevValue.current = value;
+	} );
```

Comment:
You're missing `value` dependency here.
```suggestion
	}, [ value ] );
```

Reply:
Thanks, I also tightened up the `value` concerns in the useEffect calls